### PR TITLE
feat: Rework config hash and add anonymization

### PR DESCRIPTION
**Pull request by lukas-valenta**
## Description

Creates a function to get a subset of super.json to send with metrics. Reworks config hash function to work with this subset.

## Motivation and Context

We need to send the subset to backend with metrics.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have updated the documentation accordingly. If you are changing **code related to user secrets** you need to really make sure that [security documentation](SECURITY.md) is correct.
- [x] I have read the **CONTRIBUTION_GUIDE** document.
- [x] I haven't repeated the code. (DRY)
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -11,7 +11,7 @@
     "noFallthroughCasesInSwitch": true,
     "noImplicitAny": true,
     "noImplicitReturns": true,
-    "noImplicitOverride": true,
+    "noImplicitOverride": false,
```
Comment (Jakub-Vacek) [inline tsconfig.json:14]: 😂

Comment (lukas-valenta) [reply tsconfig.json:14]: only until the prettier fix releases!

File: src/internal/superjson/schema.ts [hunk 2]
```diff
@@ -290,3 +290,18 @@ export type NormalizedProfileProviderSettings = zod.infer<
...
+
+export type AnonymizedSuperJsonDocument = {
+  profiles: Record<
+    string,
+    {
+      version: string | 'file';
+      providers: {
+        provider: string;
+        priority: number;
+        version?: string | 'file';
```
Comment (Jakub-Vacek) [inline src/internal/superjson/schema.ts:302]: This is version of map?

Comment (lukas-valenta) [reply src/internal/superjson/schema.ts:302]: i think so

File: src/internal/superjson/superjson.ts [hunk 3]
```diff
@@ -256,44 +257,74 @@ export class SuperJson {
...
-          if ('file' in info) {
-            return `${provider}:${info.file}`;
-          } else {
-            return `${provider}:${info.mapVariant ?? ''}-${
-              info.mapRevision ?? ''
-            }`;
-          }
+    return {
+      profiles,
+      providers: Object.keys(this.normalized.providers),
```
Comment (Jakub-Vacek) [inline src/internal/superjson/superjson.ts:294]: Can normalized providers contain secrets - loaded from env or directly stored in super.json

**Review by Jakub-Vacek — APPROVED** [review]

File: src/internal/superjson/superjson.ts (hunk 3 above)
Comment (lukas-valenta) [reply src/internal/superjson/superjson.ts:294]: no, this is just keys == names

**Pull request merged**